apps/cowswap-frontend/src/modules/swap/services/ethFlow/steps/signEthFlowOrderStep.ts

```diff
@@ -58,21 +58,29 @@ export async function signEthFlowOrderStep(
   }
 
   const ethTxOptions = { value: etherValue.quotient.toString() }
-  const estimatedGas = await ethFlowContract.estimateGas
-    .createOrder(ethOrderParams, { value: etherValue.quotient.toString() })
-    .catch((error) => {
-      logTradeFlowError(
-        'ETH FLOW',
-        '[EthFlow::SignEthFlowOrderStep] Error estimating createOrder gas. Using default ' + GAS_LIMIT_DEFAULT,
-        error
-      )
-      return GAS_LIMIT_DEFAULT
-    })
-
-  const txReceipt = await ethFlowContract.createOrder(ethOrderParams, {
+  const estimatedGas = await ethFlowContract.estimateGas.createOrder(ethOrderParams, ethTxOptions).catch((error) => {
+    logTradeFlowError(
+      'ETH FLOW',
+      '[EthFlow::SignEthFlowOrderStep] Error estimating createOrder gas. Using default ' + GAS_LIMIT_DEFAULT,
+      error,
+    )
+    return GAS_LIMIT_DEFAULT
+  })
+
+  // This used to be done with a higher level of abstraction like this:
+  // const txReceipt = await ethFlowContract.createOrder(ethOrderParams, {
+  //   ...ethTxOptions,
+  //   gasLimit: calculateGasMargin(estimatedGas),
+  // })
+  // However, to **try** to prevent wallet issues, we want to explicitly send along the chainId
+  // But that wrapper doesn't accept it.
+  // So we must build the tx first, then send it using the contract's signer
+  const tx = await ethFlowContract.populateTransaction.createOrder(ethOrderParams, {
     ...ethTxOptions,
     gasLimit: calculateGasMargin(estimatedGas),
   })
+  const txReceipt = await ethFlowContract.signer.sendTransaction({ ...tx, chainId: orderParams.chainId })
```

Comment:
ubernit: Alothough you asserted is the same, I would rather use before `network.chainId` instead of `orderParams.chainId`. The reason is, because this is var comes from the contract. What we try to do is ensure the contract chain is the same where we place the transaction.

Sligly prefer this, cause If i want to go back to `orderParams` definition, I see its a param passed to this helper function so I don't know its the case. If I check however where `network` is defined, I see is coming from the contract. Also, if for whatever reason someone deletes the assertion, the is harder to notice what we wanted to pass here (chain id of the order param, or the chain id from the contract)

Reply:
https://github.com/cowprotocol/cowswap/pull/5263

Reply:
👍